metrique-macro/src/lib.rs

```diff
@@ -471,12 +612,73 @@ struct MetricsFieldAttrs {
     kind: MetricsFieldKind,
 }
 
+#[derive(Debug, Clone)]
+enum Prefix {
+    Inflectable {
+        prefix: String,
+        contains_dot: Option<Span>,
+    },
+    Exact(String),
+}
+
+impl Prefix {
+    fn inflected_prefix_message(c: char, is_warning: bool) -> String {
+        let warning_text = if is_warning {
+            " `.` is currently allowed in an inflected prefix, but will be disallowed in a future version."
+        } else {
+            ""
+        };
+        format!(
+            "Found {c:?} in inflected prefix. `prefix` inflection treats all non-alphanumeric \
+                        characters identically. Use `-` or `_` for an inflectable separator, or use `exact_prefix` for \
+                        a separator that will not be inflected.{warning_text}"
```

Comment:
"Or use `exact_prefix` to preserve special characters in the metric name"?

I worry that people don't necessarily know exactly what is meant by "inflected"